bin/update-schematron.php

```diff
@@ -20,8 +27,29 @@
     'trunk/converters/code/ToSchematron/ExtractSchFromXSD.xsl',
 ];
 
-$filesystem->remove(new FilesystemIterator($target));
+if (!$filesystem->exists($repoDir)) {
+    $filesystem->mkdir($repoDir);
+    $repo = $git->cloneRepository($repoUri, $repoDir, ['depth' => 1]);
+} else {
+    $repo = $git->workingCopy($repoDir);
+    $repo->fetch('origin', 'master');
+    $repo->reset(['hard' => true], 'origin/master');
+}
+
+foreach ($patches as $patch) {
+    $patchTarget = "${patchesDir}/".md5($patch);
+
+    echo "Downloading {$patch} to {$patchTarget}\n";
+    $filesystem->copy($patch, $patchTarget);
+    $repo->apply($patchTarget, ['verbose' => true]);
```

Comment:
I think by default the patches remain tracked in Git as a change not staged for commit. The interaction with updates should work because then a `git reset --hard` will be performed.